### PR TITLE
moved mkdocs.yml to the docs directory

### PR DESCRIPTION
**Pull request by Lescurel**
- Modified the github actions workflow to work with that 
- Added `./site` to `.gitignore` to avoid unwanted files to the repo 
- Added a make rule to preview and work with the doc (it uses docker to avoid setting up a python environment)

## Summary by CodeRabbit

- **New Features**
	- Introduced a new target for building and serving documentation via Docker.
	- Updated workflow for publishing documentation to GitHub Pages, ensuring proper build and deployment processes.
	- Added a custom domain for the documentation site.

- **Bug Fixes**
	- Corrected indentation issues in the Makefile for proper execution.

- **Chores**
	- Added new dependencies for MkDocs to enhance documentation capabilities.
	- Updated `.gitignore` to exclude generated files.

- **Documentation**
	- Modified documentation configuration for improved structure and output directory.

### CODE REVIEW

File: docs/src/CNAME
Comment (emanuelen5) [inline docs/src/CNAME:1]: Not sure if you're allowed to move this file. It was auto-generated by Github actions when registering our domain through the repo settings

File: docs/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM python@sha256:5d769f990397afbb2aca24b0655e404c0f2806d268f454b052e81e39d87abf42
```
Comment (emanuelen5) [inline docs/Dockerfile:1]: You need to add the path to this Docker file here https://github.com/makerspace/makeradmin/blob/3b912abfd2cbf2263dd8b4bea6a573ffc9384a68/update-docker-images.sh#L41-L42 so it's also kept up to date by the script

File: docs/src/CNAME
Comment (Lescurel) [reply docs/src/CNAME:1]: According to [MkDocs documentation](https://www.mkdocs.org/user-guide/deploying-your-docs/#custom-domains), the `CNAME` file should be in the root of the `docs_dir`, which is `docs/src` now, so it should be fine.

File: docs/Dockerfile (hunk 1 above)
Comment (Lescurel) [reply docs/Dockerfile:1]: Done :)

File: docs/src/CNAME
Comment (emanuelen5) [reply docs/src/CNAME:1]: Ah, great! Thanks for looking up :)

**Review by emanuelen5 — APPROVED** [review]

**Pull request merged**